### ISSUE TITLE
Is there a way to use this add-on on Firefox or how to convert it to Firefox Addons?

### ISSUE THREAD

**Issue by directentis1** [benf2004/Bing-Chat-History#41]
I use both Chrome and Firefox for my workflow and it would be more consistent if I could discuss and export my Bing chats in the same place instead of having to constantly switch browsers to save only my Bing chats.

 I think it is not too complicated to switch from Chrome APIs to Firefox WebExtension APIs, can you guide or give me suggestions?
 I will keep credit of you.

**Comment by benf2004** [comment]
Yes, it's pretty easy to do, but I thought that Bing Chat did not work on Firefox (I guess you might have the Bing Chat for All Browsers Extension). 

To port it to Firefox, all you need to do is switch from manifest version 3 to manifest version 2. Firefox lets you use the `chrome` namespace but the reverse is not true - you can't use the `browser` namespace on Chrome. There are a few differences in the WebEextension APIs, but not many. The notable one is `chrome.action` should be `chrome.browserAction` namespace. 

If you do decide to port it over, I'd appreciate it if you'd make a pull request for this project :) I might take a look if I have time. You can look at the sister project, ChatGPT Prompt Genius for an example of how having both manifests works.

**Comment by imatimba** [comment]
Hey, just wanted to share that manifest version 3 works in Firefox since version 109.  
The only changes I had to make in the manifest to get it accepted in addons.mozilla.org dev hub were adding this section:  
```  
"browser_specific_settings": {
      "gecko": {
          "id": "bingchathistory@anything.com",
          "strict_min_version": "110.0"
  }},
```
And changing this section:

```
"background": {
    "scripts": ["background.js"]
},
```

After a day awaiting review it was accepted and signed, seems to work fine so far.

**Comment by ShapeShifter499** [comment]
> Yes, it's pretty easy to do, but I thought that Bing Chat did not work on Firefox (I guess you might have the Bing Chat for All Browsers Extension).

Bing AI chat works fine here on firefox and I didn't have to enable any extensions, was there a change?

**Comment by BillCampbell** [comment]
As of a couple of months ago (maybe starting in July or August), Bing Chat (recently renamed to copilot) has been available in other browsers. Back in May when the previous comment was posted, it only officially worked in Edge.
